src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java

```diff
@@ -27,21 +27,18 @@
 public class UftpSigningServiceImplementation implements UftpSigningService {
 
     private String role;
+    private LazySodiumJava lazySodium;
 
     public UftpSigningServiceImplementation(String role) {
         this.role = role;
+        this.lazySodium = new LazySodiumJava(new SodiumJava(), new Base64MessageEncoder());
     }
 
     public String sealMessage(String message, String privateKey, DomainPair domainPair) {
-        LazySodiumJava lazySodium = new LazySodiumJava(new SodiumJava());
         try {
-            var body = lazySodium.cryptoSign(message, Key.fromHexString(privateKey));
-            var base64Body = Base64.getEncoder().encodeToString(body.getBytes());
-
+            String base64Body = lazySodium.cryptoSign(message, Key.fromHexString(privateKey));
```

Comment:
You changed the expected public key format to a base64 string (this was a hex string before), this is a good change.
it's inconsistent if the private key is expected to be a hex string. So can you also change the private key?

Reply:
done.  I also included the missing change to the key generation from our internal repository.

Changes pushed to appbeats repo

Reply:
pushed change to this repo branch now